src/webgpu/api/operation/rendering/color_target_state.spec.ts

```diff
@@ -791,19 +859,36 @@ g.test('blending,clamping')
   Test that clamping occurs at the correct points in the blend process: src value, src factor, dst
   factor, and output.
     - TODO: Need to test snorm formats.
-    - TODO: Need to test src value, srcFactor and dstFactor.
   `
   )
   .params(u =>
     u //
       .combine('format', ['rgba8unorm', 'rg16float'] as const)
+      .combine('srcFactor', kBlendFactors)
+      .combine('dstFactor', kBlendFactors)
+      .filter(t => {
+        return (
+          !(
+            t.srcFactor === 'one-minus-src-alpha' ||
```

Comment:
If I run this PR without the filter then I hit an `unreachable` in `calculateExpectedClampValue`. But fixing that it still fails. At least some of the failures are because we expect the color and alpha channels to always be the same, but that's not always true: https://gpuweb.github.io/gpuweb/#enumdef-gpublendfactor

> "src-alpha-saturated" (min(Asrc, 1 - Adst), min(Asrc, 1 - Adst), min(Asrc, 1 - Adst), 1)

Note the last channel has different factors.

The other issue I see is that when the format is `rg16float`, there is no alpha channel. Therefore dst-alpha won't have the initial value the code expects. I think missing channels default to ,0,0,1 so dst-alpha would be 1 (similar to [this](https://gpuweb.github.io/gpuweb/#color-space-conversions)), but I'm not 100% sure about that.

Reply:
It's ,0,0,1. Was confirmed in https://github.com/gpuweb/gpuweb/issues/2025

Reply:
> The other issue I see is that when the format is rg16float, there is no alpha channel. Therefore dst-alpha won't have the initial value the code expects. I think missing channels default to ,0,0,1 so dst-alpha would be 1 (similar to [this](https://gpuweb.github.io/gpuweb/#color-space-conversions)), but I'm not 100% sure about that.

yes, as you thought, the failures come from `rg16float` format and `src-alpha-saturated` factor. It's likely this PR should make the test use RGBA instead of using one value.